### PR TITLE
gh-126947: Typechecking for _pydatetime.timedelta.__new__ arguments

### PR DESCRIPTION
**Pull request by bombs-kim**
* Add typecheck in _pydatetime.timedelta so that same format error is raised as in _datetime.timedelta
* Add relevant tests for the new typecheck in datetime module

* Issue: gh-126947

### CODE REVIEW

**Review by sobolevn — COMMENTED** [review]
Now python implementation is in line with C implementation. Please, add a NEWS entry.

**Comment by bombs-kim** [comment]
@sobolevn a NEWS entry has been added

File: Misc/NEWS.d/next/Library/2024-11-18-19-03-46.gh-issue-126947.NiDYUe.rst
Comment (sobolevn) [inline Misc/NEWS.d/next/Library/2024-11-18-19-03-46.gh-issue-126947.NiDYUe.rst]:
```suggestion
Raise :exc:`TypeError` in :meth:`!_pydatetime.timedelta.__new__` arguments are not :class:`int` or :class:`float`, so that python
implementation is in line with C implementation.
```

File: Misc/NEWS.d/next/Library/2024-11-18-19-03-46.gh-issue-126947.NiDYUe.rst
Comment (ZeroIntensity) [inline Misc/NEWS.d/next/Library/2024-11-18-19-03-46.gh-issue-126947.NiDYUe.rst]: Nitpicks:
```suggestion
Raise :exc:`TypeError` in :meth:`!_pydatetime.timedelta.__new__` if the passed arguments are not :class:`int` or :class:`float`, so that the Python
implementation is in line with the C implementation.
```

**Comment by sobolevn** [comment]
Thanks! Let's wait for code-owners to review :)

**Review by vstinner — APPROVED** [review]
LGTM. The PR doesn't change the exception type, but provides a better error message.

**Review by ZeroIntensity — APPROVED** [review]
I might as well hop on the approval train :)

How do we feel about backporting?

**Comment by vstinner** [comment]
> How do we feel about backporting?

It's a bad idea to change error messages in a stable release, it can break tests.

**Review by Eclips4 — APPROVED** [review]

**Comment by sobolevn** [comment]
Thanks everyone!

**Pull request merged**
